Bluepill-cli/Bluepill-cli/Simulator/SimulatorScreenshotService.m

```diff
@@ -0,0 +1,159 @@
+//  Copyright 2016 LinkedIn Corporation
+//  Licensed under the BSD 2-Clause License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at https://opensource.org/licenses/BSD-2-Clause
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OF ANY KIND, either express or implied.  See the License for the specific language governing permissions and limitations under the License.
+
+#import "SimulatorScreenshotService.h"
+#import "SimDeviceFramebufferService.h"
+#import "BPUtils.h"
+#import <CoreImage/CoreImage.h>
+
+@interface SimulatorScreenshotService()
+
+@property (nonatomic, strong) BPConfiguration *config;
+@property (nonatomic, strong) SimDevice *device;
+@property (nonatomic, strong) SimDeviceFramebufferService *frameBufferService;
+@property (nonatomic, strong) dispatch_queue_t frameBufferQueue;
+@property (nonatomic) IOSurfaceRef ioSurface;
+
+@end
+
+@implementation SimulatorScreenshotService
+
+- (instancetype)initWithConfiguration:(BPConfiguration *)config forDevice:(SimDevice *)device {
+    self = [super init];
+    if (self) {
+        _config = config;
+        _device = device;
+
+        if ([self meetsPreconditionsForConnectingToDevice:device]) {
+            _frameBufferService = [self createMainScreenServiceForDevice:device];
+            [self startService];
+        }
+    }
+
+    return self;
+}
+
+- (void)dealloc {
+    [self stopService];
+}
+
+- (CGImageRef)screenshot {
+    CIContext *context = [CIContext contextWithOptions:nil];
+    CIImage *ciImage = [CIImage imageWithIOSurface:_ioSurface];
+
+    CGImageRef cgImage = [context createCGImage:ciImage fromRect:ciImage.extent];
+    if (!cgImage) {
+        [BPUtils printInfo:ERROR withString:@"Rendering simulator screenshot failed, returning null"];
+        return NULL;
+    }
+
+    return cgImage;
+}
+
+- (void)saveScreenshotForFailedTestWithName:(NSString *)name {
+    [self saveScreenshotForFailedTestWithName:name suffix:1];
```

Comment:
suffix:0?

Reply:
first atempt -> 1 :P

Reply:
you did `suffix + 1` later in `saveScreenshotForFailedTestWithName` function

Reply:
yea, but only if file with suffix=1 already exists, so it will be:
- myTestname_attempt_1
- myTestname_attempt_2 etc.